modules/core/src/v2/coins/avaxc.ts

```diff
@@ -35,24 +352,330 @@ export class AvaxC extends AbstractEthLikeCoin {
+    if (!_.isNil(originalParams)) {
+      const { recipients } = originalParams;
+
+      // Then validate that the tx params actually equal the requested params
+      const originalAmount = new BigNumber(recipients[0].amount);
+      const originalDestination: string = recipients[0].address;
+
+      const hopAmount = new BigNumber(optionalDeps.ethUtil.bufferToHex(builtHopTx.value));
+      if (!builtHopTx.to) {
+        throw new Error(`Transaction does not have a destination address`);
+      }
+      const hopDestination = builtHopTx.to.toString();
+      if (!hopAmount.eq(originalAmount)) {
+        throw new Error(`Hop amount: ${hopAmount} does not equal original amount: ${originalAmount}`);
+      }
+      if (hopDestination.toLowerCase() !== originalDestination.toLowerCase()) {
+        throw new Error(`Hop destination: ${hopDestination} does not equal original recipient: ${hopDestination}`);
+      }
+    }
+
+    if (!builtHopTx.verifySignature()) {
+      // We dont want to continue at all in this case, at risk of ETH being stuck on the hop address
+      throw new Error(`Invalid hop transaction signature, txid: ${id}`);
+    }
+    if (optionalDeps.ethUtil.addHexPrefix(builtHopTx.hash().toString('hex')) !== id) {
+      throw new Error(`Signed hop txid does not equal actual txid`);
+    }
+  }
+
+  /**
+   * Assemble half-sign prebuilt transaction
+   * @param params
+   */
+  async signTransaction(params: SignTransactionOptions): Promise<SignedTransaction> {
     const txBuilder = this.getTransactionBuilder();
     txBuilder.from(params.txPrebuild.txHex);
     txBuilder.transfer().key(new AvaxCAccountLib.KeyPair({ prv: params.prv }).getKeys().prv!);
     const transaction = await txBuilder.build();
 
     const recipients = transaction.outputs.map((output) => ({ address: output.address, amount: output.value }));
 
+    const txParams = {
+      txHex: transaction.toBroadcastFormat(),
+      recipients: recipients,
+      expiration: params.txPrebuild.expireTime,
+      hopTransaction: params.txPrebuild.hopTransaction,
+    };
+
+    return { halfSigned: txParams };
+  }
+
+  /**
+   * Modify prebuild before sending it to the server. Add things like hop transaction params
+   * @param buildParams The whitelisted parameters for this prebuild
+   * @param buildParams.hop True if this should prebuild a hop tx, else false
+   * @param buildParams.recipients The recipients array of this transaction
+   * @param buildParams.wallet The wallet sending this tx
+   * @param buildParams.walletPassphrase the passphrase for this wallet
+   */
+  async getExtraPrebuildParams(buildParams: BuildOptions): Promise<BuildOptions> {
+    if (
+      !_.isUndefined(buildParams.hop) &&
+      buildParams.hop &&
+      !_.isUndefined(buildParams.wallet) &&
+      !_.isUndefined(buildParams.recipients) &&
+      !_.isUndefined(buildParams.walletPassphrase)
+    ) {
+      if (this instanceof Erc20Token) {
```

Comment:
are avalanche tokens an instance of Erc20Token?

Reply:
nope, i should change this